src/pull_request.ts

```diff
@@ -53,19 +47,76 @@ export async function handlePullRequest() {
   let overviewComment = existingComments.find((comment) =>
     comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
   );
+  const isIncrementalReview = !!overviewComment;
+
+  // Maybe fetch review comments
+  const reviewComments = isIncrementalReview
+    ? (
+        await octokit.rest.pulls.listReviewComments({
+          ...context.repo,
+          pull_number: pull_request.number,
+        })
+      ).data
+    : [];
+
+  // Get modified files
+  const { data: files } = await octokit.rest.pulls.listFiles({
+    ...context.repo,
+    pull_number: pull_request.number,
+  });
+  let filesToReview = files.map((file) => parseFileDiff(file, reviewComments));
+  info(`successfully fetched file diffs`);
+
+  let commitsReviewed: string[] = [];
   if (overviewComment) {
+    info(`running incremental review`);
+    try {
+      const payload = JSON.parse(
+        overviewComment.body
+          ?.split(PAYLOAD_TAG_OPEN)[1]
+          .split(PAYLOAD_TAG_CLOSE)[0] || "{}"
+      );
+      commitsReviewed = payload.commits;
+    } catch (error) {
+      warning(`error parsing overview payload: ${error}`);
+    }
+
+    // Check if there are any incremental changes
+    const lastCommitReviewed = commitsReviewed.length
+      ? commitsReviewed[commitsReviewed.length - 1]
+      : null;
+    const incrementalDiff =
+      lastCommitReviewed && lastCommitReviewed != pull_request.head.sha
+        ? await octokit.rest.repos.compareCommits({
+            ...context.repo,
+            base: lastCommitReviewed,
+            head: pull_request.head.sha,
+          })
+        : null;
```

Comment:
The commit comparison API request should be wrapped in a try-catch block to handle potential failures. Consider adding error handling:
```typescript
let incrementalDiff = null;
if (lastCommitReviewed && lastCommitReviewed !== pull_request.head.sha) {
  try {
    incrementalDiff = await octokit.rest.repos.compareCommits({
      ...context.repo,
      base: lastCommitReviewed,
      head: pull_request.head.sha,
    });
  } catch (error) {
    warning(`Failed to compare commits: ${error}`);
  }
}
```